sonaranalyzer-dotnet/src/SonarAnalyzer.CSharp/Rules/DoNotHardcodeCredentials.cs

```diff
@@ -55,7 +55,7 @@ private class VariableDeclarationBannedWordsFinder : CredentialWordsFinderBase<V
             public VariableDeclarationBannedWordsFinder(DoNotHardcodeCredentialsBase analyzer) : base(analyzer) { }
 
             protected override string GetAssignedValue(VariableDeclaratorSyntax syntaxNode) =>
-                syntaxNode.Initializer?.Value.ToString();
+                (syntaxNode.Initializer?.Value as LiteralExpressionSyntax)?.Token.ValueText;

```

Comment:
you could use the freshly added `GetStringValue` extension method from `CSharpSyntaxHelper` (has null check inside)

```suggestion
                syntaxNode.Initializer?.Value.GetStringValue();
```